src/datasets/iterable_dataset.py

```diff
@@ -587,6 +673,11 @@ def _iter_shard(self, shard_idx: int):
 
     def _apply_feature_types(self, example):
         if self.features:
+            example = dict(example)
+            # add missing columns
+            for column_name in self.features:
+                if column_name not in example:
+                    example[column_name] = None
```

Comment:
If I'm not mistaken, one can accidentally introduce new columns in `cast` and these columns' values will be returned as `None` due to this change. But I think it's OK to leave this as-is for now and add checks later for consistency with `Dataset`.